api/src/wfl/service/datarepo.clj

```diff
@@ -117,3 +118,126 @@
       util/response-body-json
       :id
       poll-job))
+
+(defn create-snapshot
+  "Create a snapshot from standard SQL query,
+   assert or row-ids, based on `snapshot-request`.
+
+   See `SnapshotRequestModel` in the
+   DataRepo swagger page for more information.
+   https://jade.datarepo-dev.broadinstitute.org/swagger-ui.html#/
+
+   Note the TDR is under active development,
+   the endpoint spec is getting changed so the
+   spec in this function is not consistent with
+   the TDR Swagger page in order to make the
+   request work."
+  [snapshot-request]
+  (-> (repository "snapshots")
+      (http/post {:headers      (auth/get-service-account-header)
+                  :content-type :application/json
+                  :form-params  snapshot-request})
+      util/response-body-json
+      :id
+      poll-job))
+
+(defn list-snapshots
+  "List all Snapshots the caller has access to.
+   Hard-coded to return 999 pages for now.
+
+   Parameters
+   ----------
+   dataset-ids      - Optionally filter the result snapshots
+                      where provided datasets are source datasets.
+
+   Example
+   -------
+     (list-snapshots)
+     (list-snapshots \"48a51f71-6bab-483d-a270-3f9ebfb241cd\" \"85efdfea-52fb-4698-bee6-eef76104a7f4\")"
+  [& dataset-ids]
+  (letfn [(maybe [m k v] (if (seq v) (assoc m k {:datasetIds v}) m))]
+    (-> (http/get (repository "snapshots")
+                  (-> {:headers (auth/get-service-account-header)
+                       :query-params {:limit 999}}
```

Comment:
I wonder if there's a way to paginate this somehow so we don't have to set arbitrary lengths. I imagine that lots of snapshots will be created over the lifetime of any given project.

Reply:
Let me look into it!

It's annoying that you have to at least hit the endpoint once to fetch the page total and page token in order to start the pagination, which makes a simple map solution more complex. I wasn't patient enough to dig into that as this PR ended up using this func.

Reply:
Based on offline discussion, will defer this work to the next PR that looks for snapshots. potential solutions could be using `lazy-seq`.

Reply:
Find examples of pagination in various namespaces.